fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/contract/annotation/Transaction.java

```diff
@@ -31,9 +39,17 @@
      * semantics
      *
      * @return boolean, default is true
+     * @deprecated Please use intent
      */
+    @Deprecated
     boolean submit() default true;
 
+    /**
+     * SUBMIT - indicates that this function is intended to be called with the 'submit' semantics
+     * EVALUATE - indicates that this is intended to be called with the 'evaluate' semantics
+     */
+    TYPE intent() default Transaction.TYPE.SUBMIT;
```

Comment:
A little off topic but much nicer!

Reply:
why it wasn't like this from the start I have no idea